src/test/javascript/spec/component/exam/participate/exercises/programming-exam-submission.component.spec.ts

```diff
@@ -86,53 +96,49 @@ describe('ProgrammingExamSubmissionComponent', () => {
         component.exercise = programmingExercise;
         fixture.detectChanges();
 
-        expect(component.repositoryIsLocked).to.equal(true);
+        expect(component.repositoryIsLocked).toBe(true);
     });
 
     it('should change state on commit', () => {
-        component.studentParticipation = studentParticipation;
+        component.studentParticipation = newParticipation();
+
+        component.onCommitStateChange(CommitState.UNDEFINED);
+        expect(component.hasSubmittedOnce).toBe(false);
 
-        const undefinedCommitState = CommitState.UNDEFINED;
-        component.onCommitStateChange(undefinedCommitState);
-        expect(component.hasSubmittedOnce).to.equal(false);
+        component.onCommitStateChange(CommitState.CLEAN);
 
-        const cleanCommitState = CommitState.CLEAN;
+        // After the first call with CommitState.CLEAN, component.hasSubmittedOnce must be now true
+        expect(component.hasSubmittedOnce).toBe(true);
 
-        component.onCommitStateChange(cleanCommitState);
-        expect(component.hasSubmittedOnce).to.equal(true);
+        component.onCommitStateChange(CommitState.CLEAN);
 
-        /**
-         * After the first call with CommitState.CLEAN, component.hasSubmittedOnce must be now true
-         */
-        component.onCommitStateChange(cleanCommitState);
-        if (component.studentParticipation.submissions) {
-            expect(component.studentParticipation.submissions[0].submitted).to.equal(true);
-            expect(component.studentParticipation.submissions[0].isSynced).to.equal(true);
-        }
+        expect(component.studentParticipation.submissions![0].submitted).toBe(true);
+        expect(component.studentParticipation.submissions![0].isSynced).toBe(true);
     });
 
     it('should desync on file change', () => {
-        component.studentParticipation = studentParticipation;
-        if (component.studentParticipation.submissions) {
-            component.studentParticipation.submissions[0].isSynced = true;
-            component.onFileChanged();
+        component.studentParticipation = newParticipation();
+
+        component.studentParticipation.submissions![0].isSynced = true;
+        component.onFileChanged();
 
-            expect(component.studentParticipation.submissions[0].isSynced).to.equal(false);
-        }
+        expect(component.studentParticipation.submissions![0].isSynced).toBe(false);
     });
 
     it('should get submission', () => {
-        component.studentParticipation = studentParticipation;
-        if (studentParticipation.submissions) {
-            expect(component.getSubmission()).to.equal(studentParticipation.submissions[0]);
-        }
+        const participation = newParticipation();
+        component.studentParticipation = participation;
+
+        expect(component.getSubmission()).toBe(participation.submissions![0]);
```

Comment:
I'm not totally sure whether the exact reference should be the same rather than the submission content should equal.

Reply:
Comparing the objects using reference equality was indeed not intended. Good catch!